content/new-routing.md

```diff
@@ -0,0 +1,446 @@
+
+### The current route
+
+To access the current route, you can use `FlowRouter.current()`. This is a object representing all aspects of the route, and as it changes often it is not reactive. 
+
+Often it's more useful to access just exactly what parts of the route you care about. Here are some useful reactive functions you can call:
+
+* `FlowRouter.getRouteName()` gets the name of the route
+* `FlowRouter.getParam(paramName)` returns the value of a single URL parameter
+* `FlowRouter.getQueryParam(paramName)` returns the value of a single URL query parameter
+
+So in our example of the list page form the Todos app, we access the current list's id with `FlowRouter.getParam('_id')` (we'll see more on this below).
+
+### Highlighting the active route
+
+One situation where it is sensible to access the global `FlowRouter` singleton to access the current route's information deeper in the component hierarchy is when rendering links via a navigation component. It's often required to highlight the "active" route in some way (this is the route or section of the site that the user is currently looking at).
+
+A convenient package for this is [`zimme:active-route`](https://github.com/zimme/meteor-active-route):
+
+```bash
+meteor add zimme:active-route
+```
+
+In the Todos example app, we link to each list the user knows about in the `appBody` template:
+
+```blaze
+{{#each list in lists}}
+  <a class="list-todo {{activeListClass list}}">
+    ...
+
+    {{list.name}}
+  </a>
+{{/each}}
+```
+
+We can determine if the user is currently viewing the list with the `activeListClass` helper:
+
+```js
+Template.appBody.helpers({
+  activeListClass(list) {
+    const active = ActiveRoute.name('listsShow')
+      && FlowRouter.getParam('_id') === list._id;
+
+    return active && 'active';
+  }
+});
+```
+
+## Rendering based on the route
+
+Now we understand how to define routes and access information about the current route, we are in a position to do you usually want to do when a user accesses a route---render a user interface to the screen that represents it.
+
+*In this section, we'll discuss how to render routes using Blaze as the UI engine. If you are building your app with React or Angular, you will end up with similar concepts but the code will not be exactly the same.*
+
+When using Flow Router, the simplest way to display different views on the page for different URLs is to use the complementary Blaze Layout package. First, make sure you have the Blaze Layout package installed:
+
+```bash
+meteor add kadira:blaze-layout
+```
+
+To use this package, we need to render a "layout" component by default. In the Todos example app, that template is called `appBody`:
+
+```blaze
```

Comment:
Yeah, I'm not sure. We could add highlighting for blaze conceptually. @yyx990803 - what do you think?